src/main/java/sirius/pasta/tagliatelle/compiler/TemplateCompiler.java

```diff
@@ -79,7 +79,16 @@ public List<CompileError> compile() throws CompileException {
         }
 
         try {
-            Emitter emitter = parseBlock(null, null).reduce();
+            // identify content type via file extension, as is done in Template.setupEscaper(GlobalRenderContext), in
+            // order to decide whether to trim whitespace after parsing
+            String effectiveFileName = getContext().getTemplate().getEffectiveFileName();
+            boolean trimWhitespace = effectiveFileName.endsWith(".html") || effectiveFileName.endsWith(".xml");
```

Comment:
Template#setupEscaper has already a similar check, maybe we could extract this as Template#isXmlContentExpected

Reply:
done